### PR TITLE
Controllers test

### PR DESCRIPTION
**Pull request by Luchy-B**
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Carried out checks

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      expect(response.body).to include('Here is a list of posts with comments for a given user')
+    end
+
+    it 'should return success' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include('Post details for a given user')
+    end
+  end
+end
```
Comment (mckent05) [inline spec/requests/posts_spec.rb]: - Good job done so far, but currently all your tests are not passing, a possible reason for this is that you are not using the right request type. Your test file should look like this:
```ruby
   require 'rails_helper'

RSpec.describe 'Posts', type: :request do
  describe 'GET/show' do
    context 'with success response' do
      it 'should return success' do
        get '/users/'
        expect(response).to have_http_status(:success)
        expect(response.body).to include('Lists of all users')
      end
  
      it 'should return success' do
        get '/users/:id'
        expect(response).to have_http_status(:success)
        expect(response.body).to include('Information about a given user')
      end
    end
  end
end
```
- Please fix the above issue for both test files 👍 👍 
srgasrga

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    it 'should return success' do
+      get '/users/:user_id/posts'
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include('Here is a list of posts with comments for a given user')
+    end
+
+    it 'should return success' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include('Post details for a given user')
```
Comment (mckent05) [inline spec/requests/posts_spec.rb]: - According to the project requirements, for each action, you should check if the correct template is rendered, but this is missing in your test. Please correct this for **all** action tests 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Luchy-B, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job testing your controllers, keep it up 🚀 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @Luchy-B  👋 

### Things to Appreciate
- [ ] Fixed issues requested by the previous Code Reviewer 👍 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Luchy-B** [comment]
Thanks for the review

**Pull request merged**